src/app/pages/onboarding/sign-in/utils/mnemonic-validation.ts

```diff
@@ -0,0 +1,24 @@
+import { getDefaultWordlist, wordlists } from 'bip39';
+import * as yup from 'yup';
+
+function isValidMnemonic(word: string): boolean {
+  return wordlists[getDefaultWordlist()].includes(word);
+}
+
```

Comment:
The field names are dynamically generated using an `index` (`mnemonic-input-${index}`) so it was tricky to correctly add per field validation